front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +59,44 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points as NonNullable<
+          PathProperties['operational_points']
+        >;
+        // Check if there are vias added by map click and insert them in the operational points
+        if (trainScheduleUsedForProjection.path.some((step) => 'track' in step)) {
+          trainScheduleUsedForProjection.path.forEach((step, i) => {
+            if ('track' in step) {
```

Comment:
To avoid growing the indentation, an early return could be used:

```ts
if (!('track' in step)) {
  return;
}
```

Reply:
Done